app/src/main/java/com/doubtless/doubtless/screens/dashboard/usecases/FetchUserDataUseCase.kt

```diff
@@ -34,61 +33,104 @@ class FetchUserDataUseCase constructor(
         class Error(val message: String) : Result()
     }
 
+    sealed class UserDetailsResult {
+        class Success(val user: User) : UserDetailsResult()
+        class Error(val message: String) : UserDetailsResult()
+
+    }
+
     private var lastDoubtData: DoubtData? = null
     private val NOT_SET = -1L
     private val FEED_ENDED = 0L
     private var collectionCount = NOT_SET
 
     private var docFetched = 0L
 
-    @WorkerThread
-    suspend fun fetchFeedSync(request: FetchUserFeedRequest): Result = withContext(Dispatchers.IO) {
+    suspend fun fetchUserDetails(request: FetchUserFeedRequest): UserDetailsResult =
+        withContext(Dispatchers.IO) {
+            try {
+                val querySnapshot = firestore.collection(FirestoreCollection.USER)
+                    .whereEqualTo("id", request.user.id)
+                    .get().await()
+
+                var userData = User()
+                querySnapshot.documents[0].let {
+                    val user = User.parse(it) ?: return@let
+                    userData = user
+
+                    val localUserAttrCollectionRef =
+                        it.reference.collection("user_attr")
+                    val userAttrQuerySnapshot = localUserAttrCollectionRef.get().await()
+
+
+                    if (!userAttrQuerySnapshot.isEmpty) {
+                        userAttrQuerySnapshot.documents.forEach { it ->
+                            val userAttr = UserAttributes.parse(it) ?: return@forEach
+                            user.local_user_attr = userAttr
+                        }
+
+                    }
+
+                }
+
+                UserDetailsResult.Success(userData)
+
 
-        // if this is a refresh call then reset all the params,
-        // though for now, we don't reset collection count.
-        if (request.fetchFromPage1) {
-            docFetched = 0
-            lastDoubtData = null
+            } catch (e: Exception) {
+                UserDetailsResult.Error(e.message.toString())
+            }
         }
+
+
+    @WorkerThread
+    suspend fun fetchFeedSync(request: FetchUserFeedRequest): Result =
```

Comment:
Again, you are already doing work on worker thread then why annotate. This annotation is to tell the user to call function in bg thread.